core/base/src/TSystem.cxx

```diff
@@ -1019,7 +1021,8 @@ const char *TSystem::DirName(const char *pathname)
       len = res.Length() + 50;
       buf = new char [len];
    }
-   strncpy(buf, res.Data(), len);
+   if (buf)
+      strncpy(buf, res.Data(), len);
```

Comment:
The fix is okay; but the implementation of `TSystem::DirName()` looks completely wrong to me, e.g. why is it depending on a the `buf` function-static variable? @pcanal

Reply:
This is because (for better or worse) the function return a `const char*` and the author chose to resolve this by using a static.  However the code was incorrectly upgrade to be (try to be) thread safe.  Here `len` and `buf` should become `thread_local`.  Could @ferdymercury open another PR to fix this?